### ISSUE TITLE
`command_exists` not found when loading shell

### ISSUE THREAD

**Issue by drich10** [Bash-it/bash-it#1009]
As of 3fac1c7ea5b3ce4d0a270f8bea0e516278bc7473 whenever I load a new shell I see
```
-bash: command_exists: command not found
-bash: command_exists: command not found
```
This came in as a part of #974. I'm not very familiar with how load order actually happens otherwise I would submit a PR to fix myself. 

Thanks for the great tool 👍!

**Comment by nwinkler** [comment]
Thanks for reporting this! Can you please try the following and then post the output?

```bash
# First reload the config
reload

# Try to migrate to the new structure
bash-it migrate

# Then reload the config again
reload
```

If that does not fix the issue, then try this:

```bash
bash-it disable plugin base

bash-it enable plugin base

reload
```

Let me know if that fixes it - you might also want to open a new terminal window/shell.

Since the `command_exists` function is used in many places, I thought about moving it to a better location anyway. Might be time to do that now...

**Comment by tylerphelan** [comment]
Thanks for the help! The second solution fixed the problem:

```
bash-it disable plugin base

bash-it enable plugin base

reload
```

**Comment by haruka** [comment]
Hat the same problem right now. I thing one problem/reason is, that you can skip enabling the base plugin when installing via interactive mode. That was my fault right now.

**Comment by amno1** [comment]
I just got this problem myself. I have enabled some new aliases and after that I got this error. Second solution helped (disable/enable base). Worth to mention is that I got error when I disabled base plugin: "sorry, base does not appear to be an enabled plugin". But than efter issuing enable command, bash-it works fine. So some minor cosmetic bugs are in works there :-).

**Comment by nwinkler** [comment]
Thanks for reporting this - point taken 😄 

I'll refactor/move the `command_exists` function so that there's no hidden dependency. I'll do this as part of #944, where I have refactored the loading some more, and have added lots of tests as well. I'll add tests for the refactored/moved function to make sure that it works and that it does not break any existing functionality.

**Comment by nwinkler** [comment]
I've taken a stab at this, and it should be working now: https://github.com/nwinkler/bash-it/commit/e8c4063d899a783a16c4e0af66bc20b7e6b14fbe

I'll try to merge this code ASAP in a new PR...

**Comment by nwinkler** [comment]
Fixed this as part of #1043 - will wait a couple of days before merging it.

**Comment by nwinkler** [comment]
This should now be fixed, with #1043 merged. Please reopen if you still see issues.
